Code/Framework/AzFramework/AzFramework/Network/IRemoteTools.h

```diff
@@ -0,0 +1,240 @@
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+        RemoteToolsMessage(AZ::u64 msgId)
+            : m_msgId(msgId)
+            , m_senderTargetId(0)
+            , m_customBlob(nullptr)
+            , m_customBlobSize(0)
+            , m_isBlobOwner(false)
+            , m_immediateSelfDispatch(false)
+        {
+        }
+
+        virtual ~RemoteToolsMessage();
+
+        //! Add a custom data blob to a Remote Tools message
+        //! @param blob Pointer to the start of the data blob
+        //! @param blobSize Size of the data blob
+        //! @param ownBlob Whether this message owns the life cycle of the blob memory
+        void AddCustomBlob(const void* blob, size_t blobSize, bool ownBlob = false);
+
+        //! Sets if this message can be immediately dispatched if the receipient is itself
+        //! @param immediateSelfDispatchEnabled Whether this message can be immediately dispatched
+        void SetImmediateSelfDispatchEnabled(bool immediateSelfDispatchEnabled);
+
+        //! Gets if this message can be immediately dispatched if the receipient is itself
+        //! @return Whether this message can be immediately dispatched
+        bool IsImmediateSelfDispatchEnabled() const;
+
+        //! Gets the custom data blob assosciated with this message
+        //! @return Memory address of the start of the blob or nullptr if there is none
+        const void* GetCustomBlob() const;
+
+        //! Gets the size of the custom data blob associated with this message
+        //! @return Size of the data blob or 0 if there is none
+        size_t GetCustomBlobSize() const;
+
+        //! Gets the numeric ID of this message
+        //! @return The ID of this message
+        AZ::u64 GetId() const;
+
+        //! Gets this message's sender's ID
+        //! @return ID of the message's sender
+        AZ::u32 GetSenderTargetId() const;
+
+        //! Sets the ID of this message's sender
+        //! @param senderTargetId The ID to set for this message's sender
+        void SetSenderTargetId(AZ::u32 senderTargetId);
+
+    protected:
+        AZ::u64 m_msgId;
+        AZ::u32 m_senderTargetId;
+        const void* m_customBlob;
+        AZ::u32 m_customBlobSize;
+        bool m_isBlobOwner;
+        bool m_immediateSelfDispatch;
+    };
+
+    using RemoteToolsMessagePointer = AZStd::intrusive_ptr<RemoteToolsMessage>;
+    using RemoteToolsMessageQueue = AZStd::deque<RemoteToolsMessagePointer, AZ::OSStdAllocator>;
+    
+    // id for the local application
+    static constexpr AZ::u32 s_selfNetworkId = 0xFFFFFFFF;
+
+    class RemoteToolsEndpointInfo final
+    {
+
+    public:
+        AZ_CLASS_ALLOCATOR(RemoteToolsEndpointInfo, AZ::OSAllocator, 0);
+        AZ_TYPE_INFO(RemoteToolsEndpointInfo, "{DD0E9B2A-3B25-43B1-951E-CACCEC5D6754}");
+
+        RemoteToolsEndpointInfo(const AZStd::string& displayName = "", AZ::u32 networkId = 0)
+            : m_displayName(displayName)
```

Comment:
This should accept the displayName by value and move it into the m_displayName variable for efficiency.
This would cause an unnecessary copy to be performed when a string is passed normally.
i.e
```

RemoteToolsEndpointInfo endpointInfo("foo); 
// Has to construct an AZStd::string in order to bind to const AZStd::string&
// Then the constructor of RemoteToolsEndpointInfo constructor invokes the copy constructor on the m_displayName variable
```

```suggestion
        explicit RemoteToolsEndpointInfo(AZStd::string displayName = AZStd::string{},  AZ::u32 networkId = 0)
            : m_displayName(AZStd::move(displayName))
```